### PR TITLE
Feature/Setup Viewport Component

### LINKED ISSUE #214: [2] Setup Viewport Component

### LINKED ISSUE #233: Firefox does not support webgl in headless mode
Relates to #214

### PR DESCRIPTION
**Pull request by Erik9519**
#### Deployment instructions
None

#### What have I done
Setup of the view-port (Disregard the name of the branch as it is wrong)

#### Testing
Junit testing for the dispatcher. Try Exhibit on your computer and see that the canvas is properly sized.

fixes #233 
fixes #214

### CODE REVIEW

**Comment by Dacaspex** [comment]
We will test this with the new Travis configuration on Monday, see #234

File: src/app/viewport/viewport.component.ts [hunk 1]
```diff
@@ -1,16 +1,117 @@
...
+    /**
+     * ThreeJS Camera
+     */
+    private camera: THREE.Camera;
+    /**
+     * Stats.js component
+     */
+    private stats: STATS;
+
+    constructor(private dispatcher: DispatcherService) {
```
Comment (marijnvanderhorst) [inline src/app/viewport/viewport.component.ts]: This introduces a two way dependency between the Dispatcher and the Viewport, which is not desired. I see why you cannot have dispatch depend on viewport, so maybe turn the dependency around.

File: src/app/viewport/viewport.component.ts [hunk 2]
```diff
@@ -1,16 +1,117 @@
...
+    }
+
+    /**
+     * Initializes the Stats component
+     */
+    private initializeStats(): void {
+        this.stats = new STATS();
+        // Change the default stats style to absolute position (instead of relative) and z-index to 1
+        // (instead of 10000)
+        this.stats.dom.style.cssText = 'position:absolute;top:0;left:0;cursor:pointer;opacity:0.9;z-index:0';
```
Comment (marijnvanderhorst) [inline src/app/viewport/viewport.component.ts]: Comment says 1, but code says 0. Also, please add a comment why this is needed.

File: src/app/viewport/viewport.component.html [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<div class="viewport-content">
-
+<div #Container (resized)="onResize($event)" class="viewport-content w-100 h-100 flex-shrink-1">
+    <canvas #Canvas style="position: fixed;"></canvas>
```
Comment (marijnvanderhorst) [inline src/app/viewport/viewport.component.html]: Use CSS class instead

File: src/app/viewport/viewport.component.ts (hunk 1 above)
Comment (Erik9519) [reply src/app/viewport/viewport.component.ts]: Uses Event Emitter now

File: src/app/viewport/viewport.component.ts (hunk 2 above)
Comment (Erik9519) [reply src/app/viewport/viewport.component.ts]: Fixed and added short explanation.

File: src/app/viewport/viewport.component.html (hunk 3 above)
Comment (Erik9519) [reply src/app/viewport/viewport.component.html]: Now uses CSS

File: src/app/viewport/viewport.component.scss [hunk 4]
```diff
@@ -10,3 +10,7 @@
     bottom: 0;
     width:100%;
 }
+
+.viewportCanvas {
```
Comment (Dacaspex) [inline src/app/viewport/viewport.component.scss]: For consistency, we use `kebab-case` for css selectors

Comment (Erik9519) [reply src/app/viewport/viewport.component.scss]: Fixed mistake

**Pull request merged**